documentation/gems.markdown

```diff
@@ -88,7 +88,9 @@ settings with Puppet. You can manage Puppet Server's gem dependencies with the
 package provider shipped in [`puppetlabs-puppetserver_gem`](https://forge.puppet.com/puppetlabs/puppetserver_gem)
 module.
 
-## Installing Gems for use with development:
+> Note: If you try to load a gem before it's been installed, the agent run will fail with a `LoadError`.
```

Comment:
And you will need to reload the server to resolve the issue.

Reply:
Why do you only need to reload the server if you tried to load the gem before it was installed? Does something get cached?

Reply:
Yeah, that was the guy's issue that triggered this. The load error gets cached and it won't try to reload it again until a new JRuby gets it.